CXX_SUPPORTS_WE4062 ONLY for MSVC

abaelhe opened the pull request:
it breaks MacOS Swift Toolchain build: 
    $ ./swift/utils/build-toolchain com.abael

surrounding as:
if(MSVC)
    CXX_SUPPORTS_WE4062 case ...
endif()

<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves SR-NNNN.

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

abaelhe committed:
CXX_SUPPORTS_WE4062 ONLY for MSVC

it breaks MacOS Swift Toolchain build: 
    $ ./swift/utils/build-toolchain com.abael

surrounding as:
if(MSVC)
    CXX_SUPPORTS_WE4062 case ...
endif()

compnerd commented:
This would reduce the ability to detect issues with clang.  I wonder how this doesn't show up on the CI builders.

abaelhe committed:
ONLY Windows OS flags starts with leading "/"

MacOS, Linux and *BSD OSs flags ALWAYS starts with leading "-";
run clang with compiler flag "/we4062" OR "-we4062" will raise a error:

(sci) Abaels-MBP:Dockers abaelhe$ clang /we4062
clang: error: no such file or directory: '/we4062'
clang: error: no input files
(sci) Abaels-MBP:Dockers abaelhe$ clang -we4062
clang: error: unknown argument: '-we4062'
clang: error: no input files
(sci) Abaels-MBP:Dockers abaelhe$ clang -v
Apple clang version 11.0.3 (clang-1103.0.32.29)
Target: x86_64-apple-darwin19.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

References:
0.https://forums.swift.org/t/error-compiling-swift-with-ninja/37242
1.https://forums.swift.org/t/swift-build-script-failed-clang-error-no-such-file-or-directory-we4062/21127

abaelhe committed:
correction

IF -> if
